lib/sqlalchemy/sql/elements.py

```diff
@@ -3112,7 +3173,13 @@ def __init__(
     def _select_iterable(self) -> _SelectIterable:
         return (self,)
 
-    def _bind_param(self, operator, obj, type_=None, expanding=False):
+    def _bind_param(
+        self,
+        operator: OperatorType,
+        obj: List[typing_Tuple[int, str]],
+        type_: Optional[_TypeEngineArgument[_T]] = None,
+        expanding: bool = False,
+    ) -> BindParameter[_T]:
```

Comment:
- Passes `self.type` as `type_` parameter to `BindParameter.__init__`. `self.type` is of `TupleType`, which is not supported by the constructor, according to the author's type hints. Best not to touch this.
- The body of the function returns either a `BindParameter[_T]` or a `Tuple` (not to be confused with `typing.Tuple`), while superclass definition - and all overrides - only return `BindParameter[_T]`. Looks odd. Tried putting supertypes in return - that's incompatible with supertype.

Reply:
Strange: `TupleType` inherits from `TypeEngine[Tuple[Any, ...]]` and `_TypeEngineArgument` is simply a variable with `Union[Type["TypeEngine[_T]"], "TypeEngine[_T]"]` inside.